research/object_detection/core/target_assigner.py

```diff
@@ -1897,3 +1912,219 @@ def assign_corner_offset_targets(
 
     return (tf.stack(corner_targets, axis=0),
             tf.stack(foreground_targets, axis=0))
+
+class DETRTargetAssigner(object):
+  """Target assigner to compute classification and regression targets."""
+
+  def __init__(self,
+               similarity_calc,
+               matcher,
+               negative_class_weight=1.0):
+    """Construct Object Detection Target Assigner.
+
+    Args:
+      similarity_calc: a RegionSimilarityCalculator
+      matcher: an object_detection.core.Matcher used to match groundtruth to
+        anchors.
+      box_coder_instance: an object_detection.core.BoxCoder used to encode
+        matching groundtruth boxes with respect to anchors.
+      negative_class_weight: classification weight to be associated to negative
+        anchors (default: 1.0). The weight must be in [0., 1.].
+
+    Raises:
+      ValueError: if similarity_calc is not a RegionSimilarityCalculator or
+        if matcher is not a Matcher or if box_coder is not a BoxCoder
+    """
+    if not isinstance(similarity_calc, sim_calc.RegionSimilarityCalculator):
+      raise ValueError('similarity_calc must be a RegionSimilarityCalculator')
+    if not isinstance(matcher, mat.Matcher):
+      raise ValueError('matcher must be a Matcher')
+    self._similarity_calc = similarity_calc
+    self._matcher = matcher
+    self._negative_class_weight = negative_class_weight
+
+  def assign(self,
+             anchors,
+             groundtruth_boxes,
+             groundtruth_labels=None,
+             unmatched_class_label=None,
```

Comment:
I don't think we need unmatched_class_label for DETR.

Reply:
Why is that? Should I hard code it to [0] + [1]*num_classes inside target assigner?

Reply:
Needed to keep compatible with batch_assign_targets so decided to leave this in, is that ok?